securedrop/tests/test_source.py

```diff
@@ -74,462 +72,489 @@ def _find_codename(self, html):
-                codename = new_codename(client, session)
-
-                # set the expiration to ensure we trigger an expiration
-                config.SESSION_EXPIRATION_MINUTES = -1
-
-                resp = client.post('/login',
-                                   data=dict(codename=codename),
-                                   follow_redirects=True)
-                assert resp.status_code == 200
-                resp = client.get('/lookup', follow_redirects=True)
-
-                # check that the session was cleared (apart from 'expires'
-                # which is always present and 'csrf_token' which leaks no info)
-                session.pop('expires', None)
-                session.pop('csrf_token', None)
-                assert not session, session
-                assert ('You have been logged out due to inactivity' in
-                        resp.data.decode('utf-8'))
-        finally:
-            if has_session_expiration:
-                config.SESSION_EXPIRATION_MINUTES = old_expiration
-            else:
-                del config.SESSION_EXPIRATION_MINUTES
-
-    def test_csrf_error_page(self):
-        old_enabled = self.app.config['WTF_CSRF_ENABLED']
-        self.app.config['WTF_CSRF_ENABLED'] = True
-
-        try:
-            with self.app.test_client() as app:
-                resp = app.post(url_for('main.create'))
-                self.assertRedirects(resp, url_for('main.index'))
-
-                resp = app.post(url_for('main.create'), follow_redirects=True)
-                self.assertIn('Your session timed out due to inactivity',
-                              resp.data)
-        finally:
-            self.app.config['WTF_CSRF_ENABLED'] = old_enabled
+        with source_app.test_client() as app:
+            resp = app.post('/login', data=dict(codename=invalid_codename),
+                            follow_redirects=True)
+            assert resp.status_code == 200
+            text = resp.data.decode('utf-8')
+            assert "Invalid input." in text
+
+    def test_source_session_expiration(self, config, source_app):
+        with source_app.test_client() as app:
+            codename = new_codename(app, session)
+
+            # set the expiration to ensure we trigger an expiration
+            config.SESSION_EXPIRATION_MINUTES = -1
+
+            resp = app.post('/login',
+                            data=dict(codename=codename),
+                            follow_redirects=True)
+            assert resp.status_code == 200
+            resp = app.get('/lookup', follow_redirects=True)
+
+            # check that the session was cleared (apart from 'expires'
+            # which is always present and 'csrf_token' which leaks no info)
+            session.pop('expires', None)
+            session.pop('csrf_token', None)
+            assert not session, session
+            text = resp.data.decode('utf-8')
+            assert 'Your session timed out due to inactivity' in text
```

Comment:
Why is it different from `You have been logged out due to inactivity` ? It indicates the test does not behave in the same way and this is either cause for concern or a detailed explanation in the commit message.

Reply:
That string never appears in the source app, and also that test was prefixed with `_test` so it was never actually run before. We have higher coverage now that it is actually run.